tests/common/fixtures/consistency_checker/consistency_checker.py

```diff
@@ -0,0 +1,335 @@
+DUT_SCRIPT_PATH_DST_CONTAINER = DUT_DST_PATH_CONTAINER + "/" + QUERY_ASIC_SCRIPT
+
+
+class ConsistencyChecker:
+
+    def __init__(self, duthost, libsairedis_download_url=None, python3_pysairedis_download_url=None):
+        """
+        If the libsairedis_download_url and python3_pysairedis_download_url are provided, then these artifacts
+        are downloaded and installed on the DUT, otherwise it's assumed that the environment is already setup
+        for the consistency checker.
+        """
+        self._duthost = duthost
+        self._libsairedis_download_url = libsairedis_download_url
+        self._python3_pysairedis_download_url = python3_pysairedis_download_url
+
+    def __enter__(self):
+        logger.info("Initializing consistency checker on dut...")
+
+        self._duthost.shell(f"mkdir -p {DUT_DST_PATH_HOST}")
+        self._duthost.copy(src=DUT_SCRIPT_PATH_SRC, dest=DUT_SCRIPT_PATH_DST_HOST)
+
+        if self._libsairedis_download_url is not None:
+            self._duthost.shell(f"curl -o {DUT_DST_PATH_HOST}/{LIBSAIREDIS_DEB} {self._libsairedis_download_url}")
+        if self._python3_pysairedis_download_url is not None:
+            self._duthost.shell(
+                f"curl -o {DUT_DST_PATH_HOST}/{PYTHON3_PYSAIREDIS_DEB} {self._python3_pysairedis_download_url}")
+
+        # Move everything into syncd container
+        self._duthost.shell((
+            f"docker cp {DUT_DST_PATH_HOST} {SYNCD_CONTAINER}:/ && "
+            f"rm -rf {DUT_DST_PATH_HOST}"
+        ))
+
+        if self._python3_pysairedis_download_url is not None:
+            # Install python3-sairedis in syncd container
+            self._duthost.shell((f"docker exec {SYNCD_CONTAINER} bash -c "
+                                 f"'cd {DUT_DST_PATH_CONTAINER} && "
+                                 f"dpkg --install {DUT_DST_PATH_CONTAINER}/{PYTHON3_PYSAIREDIS_DEB}'"))
+
+        if self._libsairedis_download_url is not None:
+            # Extract the libsairedis deb to be used by the query script
+            self._duthost.shell((f"docker exec {SYNCD_CONTAINER} bash -c "
+                                 f"'cd {DUT_DST_PATH_CONTAINER} && "
+                                 f"dpkg --extract {DUT_DST_PATH_CONTAINER}/{LIBSAIREDIS_DEB} libsairedis-temp'"))
+
+        logger.info("Consistency checker setup complete.")
+
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        logger.info("Cleaning up consistency checker on dut...")
+
+        if self._python3_pysairedis_download_url is not None:
+            # Uninstall python3-sairedis in syncd container
+            self._duthost.shell(f"docker exec {SYNCD_CONTAINER} dpkg --remove python3-pysairedis")
+
+        # Remove all the files from the syncd container
+        self._duthost.shell(f"docker exec {SYNCD_CONTAINER} rm -rf {DUT_DST_PATH_CONTAINER}")
+
+        logger.info("Consistency checker cleanup complete.")
```

Comment:
If the consistency checker is used to do anything other than GET operations (eg., is scenario recreation), then it is better to restart syncd or do load-minigraph of the DUT as part of cleanup.

Reply:
Yeah, that's a likely direction that consistency checker will head. I'll add a comment in the code, so it's not forgotten about. Thanks